src/content-helper/excerpt-generator/components/excerpt-panel.tsx

```diff
@@ -61,24 +79,42 @@ const PostExcerptGenerator = () => {
 		};
 	}, [] );
 
-	const hasGeneratedExcerpt = generatedExcerpt.length > 0;
-	const wordCount = count( generatedExcerpt || excerpt, 'words', {} );
-	const wordCountString = sprintf(
-		// Translators: %1$s the number of words in the excerpt.
-		_n( '%1$s word', '%1$s words', wordCount, 'wp-parsely' ),
-		wordCount
-	);
+	// Update the word count string when the excerpt changes.
+	useEffect( () => {
+		/**
+		 * Returns a descriptive text for the textarea's word count.
+		 *
+		 * @since 3.16.0
+		 *
+		 * @return {string} The word count string.
+		 */
+		const getWordCountString = (): string => {
+			const wordCount = count( excerptData.currentExcerpt || excerpt, 'words', {} );
+
+			if ( wordCount > 0 ) {
+				return sprintf(
+					// Translators: %1$s the number of words in the excerpt.
+					_n( '%1$s word', '%1$s words', wordCount, 'wp-parsely' ),
+					wordCount
+				);
+			}
+
+			return '';
+		};
+
+		setWordCountString( getWordCountString() );
+	}, [ excerptData.currentExcerpt, excerpt ] );
```

Comment:
Although this is a pretty valid point, I don't think it would be an issue in this case, since it's restricted to the excerpt textarea, and adding a debounce would prevent the word count to update in real-time - unless the debounce interval would be set to very low, but that kinda defeats the purpose. 

I think we can safely ignore this suggestion. What do you think @acicovic?

Reply:
Oh, we posted at the same time 🙂

That was what I was pondering about. This might make a slight difference in some cases, but I think it's marginal, unless we set that to 1 second or something. Let's ignore this suggestion.

@coderabbitai resolve